### DISCUSSION TITLE
google apis (access users google drive)

### DISCUSSION THREAD

**Discussion by gokulgseven** [pocketbase/pocketbase#4765] [Q&A]
I want to access a google drive folder of the user and create a text file inside it .
can someone guide me how to do this , in  JavaScript  
i know the full process on how this can be done through the google apis (using the google npm package) in express.js 

what i am having trouble is with getting the npm package to run in pocketbase  ?
(don't have any knowledge of go)

**Comment by jes-ale** [comment]
Hey there! Are you using PocketBase's JS framework for this? I'm not too familiar with PocketBase or Go, but I have a suggestion. Instead of that, why not try using what you already know (Node.js with Express.js)? It's pretty straightforward. You can set up a basic server, maybe on the same machine as your PocketBase or on a different one. Then, you can talk to your server using the Fetch API and HTTP. And for a bit of security, you can use basic authentication to protect your endpoints. I know it might not be the cleanest or the most robust approach, but sometimes you just need to get the job done, right?

**Reply by gokulgseven** [reply]
i already tried something like this. I installed a express running parallel to pocketbase inside the same docker container , used nginx to route between the two using a /express path. 
but  its not a clean approach -> and too cumbersome to maintain. i dont think i will be able to maintain it in the long run , will have to probably switch back to a pure express backend (but then i loose everything i liked about pocketbase) 

i also though about building something similar to pocketbase in express , if someone wants to work with me on this shoot me a email at gokul@gseven.in , ( aim: would be to create a clone of pocketbase on express)

i want to use the pocketbase js framework for this ?

**Comment by HiImJulien** [comment]
(Sorry for necroing this discussion; I just happened to come across this.)

> what i am having trouble is with getting the npm package to run in pocketbase ?

NPM Packages are usually developed against the Node.js runtime. Which means you would require Node.js (or a compatible runtime such as Bun or Deno) to use it. The JS runtime in PB is *NOT* compatible with Node.js, which is why you're having issues getting arbitrary NPM packages to within Pocketbase.

You can either create a micro-service as suggest by @jes-ale *OR* use Pocketbase as a Go Framework. (Give it a try! If you already know JS, you'll easily pick up Go)
